packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -349,6 +349,12 @@ export interface BasicCrawlerOptions<Context extends CrawlingContext = BasicCraw
      */
     respectRobotsTxtFile?: boolean;
 
+    /**
+     * When a request is skipped for some reason, you can use this callback to act on it.
+     * This is currently fired only for requests skipped based on robots.txt file.
+     */
+    onSkippedRequest?: (request: Request | RequestOptions, reason: 'robotsTxt') => void | Promise<void>;
```

Comment:
I'm not sure about the `Request | RequestOptions` union... If I was doing this, I'd probably pick just a handful of properties present in both (`url`, `uniqueKey`, ...) so that it's easier to avoid breaking compatibility. Of course, this might make the hook less useful. What do you think?

Reply:
Hm... `{ url: string, uniqueKey: string, id?: string}` would still be sensible IMO, and you could do a `requestQueue.getRequest(id)` if you _really_ want to know more.

Reply:
Single object sure sounds nice.